mailpoet/lib/API/JSON/v1/Services.php

```diff
@@ -233,15 +239,22 @@ public function sendCongratulatoryMssEmail() {
     }
 
     $authorizedEmails = $this->bridge->getAuthorizedEmailAddresses();
-    if (!$authorizedEmails) {
+
+    $verifiedDomains = $this->senderDomainController->getVerifiedSenderDomains(true);
+
+    if (!$authorizedEmails && !$verifiedDomains) {
       return $this->createBadRequest(__('No FROM email addresses are authorized.', 'mailpoet'));
     }
```

Comment:
I think we could skip this `'No FROM email addresses are authorized.'` check. Because of the condition `!$authorizedEmails && !$verifiedDomains`, we need to always make two API requests. I think it would be nicer to take the same approach as in JS and check domains first, and it there are no domains, then check addresses or the other way around. This could save us a request in some casese. What do you think?

Reply:
Fixed in https://github.com/mailpoet/mailpoet/pull/4363/commits/236b75eddf091510f9beaf12e3923629cced2ace